xml/hwbp_registers.xml

```diff
@@ -276,14 +276,21 @@
             and writing it has no effect.
         </field>
         <field name="select" bits="19" access="WARL" reset="0">
+            This bit determines the contents of the XLEN-bit value to be
+            compared.
+
             0: Perform a match on the lowest virtual address of the access.  In
             addition, it is recommended that the trigger also matches if any of
             the other accessed virtual addresses match.
             (E.g. on a 32-bit read from 0x4000, the lowest address is 0x4000
             and the other addresses are 0x4001, 0x4002, and 0x4003.)
+            If the virtual address is smaller than DXLEN, the high bits
+            contain 0.
 
             1: Perform a match on the data value loaded or stored, or the
             instruction executed.
+            In the value to compare, any bits beyond the size of the data
+            access will contain 0.
```

Comment:
Similar thoughts here.  If DXLEN=64 and MXLEN=32 and you write tdata2 from M mode with tdata2[31]=1 then you can never match on a 32b load since the load data will have [63:32]=0 but tdata2[63:32]=0xFFFFFFFF.

Reply:
I think that's also solved by limiting matches to looking at the low XLEN bits of everything.